library/core/class.oauth2.php

```diff
@@ -220,13 +222,15 @@ public function setRequestAccessTokenParams($params) {
 
 
     /**
-     * Set additional params to be added to the get string in the getProfile request.
+     * Set additional params to be to be merged with the default parameters
+     * in the getProfile request.
+     *
+     * @param array $params Params to add to the get profile request.
      *
-     * @param string $params.
      * @return $this Return this object for chaining purposes.
      */
-    public function setGetProfileParams($params) {
-        $this->getProfileParams = $params;
+    public function setRequestProfileParams($params = []) {
```

Comment:
This is my new code where this variable is involved:

~~~
protected $requestProfileParams = [];

// (...)

public function setRequestProfileParams($params = []) {
    $this->requestProfileParams = $params;
    return $this;
}

// (...)

$requestParams = array_merge($defaultParams, $this->requestProfileParams);
~~~

Right from the start the default value for requestProfileParams is an empty array.

If the setter method is called like that `$obj->setRequestProfileParams();`, I would expect that the variable is null afterwards, though I'm not sure about that. But I'm sure it wouldn't be an empty array or any array at all. And that's a problem because the variable is used inside an `array_merge()` where the arguments need to be arrays.

Because of that it might be even better to ensure that the value is an array. What do you think about the following variant?

~~~
public function setRequestProfileParams($params) {
    $this->requestProfileParams = (arr)$params;
    return $this;
}
~~~